Cargo.toml

```diff
@@ -65,6 +65,8 @@ rstest = "0.21"
 tempfile = "3.10.1"
 tower = { version = "0.4", features = ["util"] }
 http-body-util = "0.1.1"
+testcontainers = { version = "0.21.1", features = ["watchdog"] }
+backon = "0.4.4"
```

Comment:
Nice addition! I think I can refactor the unit test of the certificate reload to make use of it

Reply:
Indeed! Very cool